src/serializers/fields.rs

```diff
@@ -229,6 +208,124 @@ impl GeneralFieldsSerializer {
         }
     }
 
+    fn sort_dict_recursive<'py>(py: Python<'py>, value: &Bound<'py, PyAny>) -> PyResult<Bound<'py, PyAny>> {
```

Comment:
I know this was suggested in a previous review, but instead of doing this recursively here, would it be an option to implement single-level sorting in this file _and_ in `src/serializers/type_serializers/dict.rs`, so that it works on any dict serialization, not just when nested in a model/dataclass/typeddict?

That way, we may not need this recursive helper function anymore, and possibly can do away with `process_field_entry_python` entirely, since `items.sort_by(|(a, _, _), (b, _, _)| a.cmp(b))` is already happening in `main_serde_serialize` and `main_to_python`.

Reply:
Thanks for guiding me.
I did a refactor based on your advice!
However, i still have to add a few functions to ensure that we dont repeat our code too much.